torch_geometric/loader/neighbor_loader.py

```diff
@@ -236,6 +236,7 @@ def __init__(
                                 collate_fn=self.neighbor_sampler, **kwargs)
 
     def transform_fn(self, out: Any) -> Union[Data, HeteroData]:
+        data = None
```

Comment:
Does it make sense for the `out` to be None? Should we perhaps not fail silently in this case?

If it does make sense then the return type of this function should now be `Optional[Union[Data, HeteroData]]`

Reply:
Thanks for the catch. Added an additional `assert` statement to ensure this.